internal/inspect/detectors/all.go

```diff
@@ -93,7 +93,11 @@ func (t *ContentTypeDetector) InferType(base util.AbsolutePath, entrypoint util.
 		} else if !aIsPreferred && bIsPreferred {
 			return 1
 		} else {
-			return strings.Compare(entrypointA, entrypointB)
+			if entrypointA == entrypointB {
+				return strings.Compare(string(a.Type), string(b.Type))
```

Comment:
This looks like a alphanumeric string comparison so this PR just sorts the configuration types on that axis rather than actually prioritizing Quarto > RMD right?

Reply:
Yes. The two cases we needed covered happened to be alphabetically ordered. I can map them to priority numbers and compare those, if this seems like too much of a shortcut.

Reply:
Totally fine waiting on an actual map of priority until it is actually needed.

My fear here is that the code here doesn't represent our intention of prioritization so it is a bit "undocumented" other than this issue.